llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -956,6 +956,9 @@ InstructionCost RISCVTTIImpl::getCastInstrCost(unsigned Opcode, Type *Dst,
     return getRISCVInstructionCost(Op, DstLT.second, CostKind);
   }
   case ISD::TRUNCATE:
+    // Early return for invalid operation
+    if (Dst->getScalarSizeInBits() >= Src->getScalarSizeInBits())
+      break;
     if (Dst->getScalarSizeInBits() == 1) {
```

Comment:
Don't repeatedly call getScalarSizeInBits on the same object. Use a variable.

Reply:
Updated. Thanks!